terraformInstall.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+
+## CONDITIONAL 1
+echo "Installing prerequisites..."
+sudo apt-get update && sudo apt-get install -y gnupg software-properties-common
+
+#!/bin/bash
+
+# Check if gpg is installed
+if ( which gpg > /dev/null 2>&1 )
+then
+    echo "gpg is installed."
+else
+    echo "Error: gpg is not installed. Please install it before running this script."
+    exit 1
+fi
+
+# Check if the keyring file exists using stat
+if (stat /usr/share/keyrings/hashicorp-archive-keyring.gpg > /dev/null 2>&1)
+then
+    echo "HashiCorp GPG key already exists."
+else
+    echo "HashiCorp GPG key not found. Adding HashiCorp GPG key."
+
+    # Add the HashiCorp GPG key
+    wget -O- https://apt.releases.hashicorp.com/gpg |
+    gpg --dearmor |
+    sudo tee /usr/share/keyrings/hashicorp-archive-keyring.gpg > /dev/null
+fi
+
+
+## CONDITIONAL 3
+echo "Verifying GPG key..."
+gpg --no-default-keyring \
```

Comment:
put the rest of these in IF statements?

Reply:
Mo, Devin and I split the work by conditionals..it is one of the reasons this project is taking so long and not straightforward- it's kind of difficult to test individual code blocks without the contribution of everything all at once.